README.md
[![Angular Logo](https://www.vectorlogo.zone/logos/angular/angular-icon.svg)](https://angular.io/) [![Electron Logo](https://www.vectorlogo.zone/logos/electronjs/electronjs-icon.svg)](https://electronjs.org/)

[![Travis Build Status][build-badge]][build]
[![Dependencies Status][dependencyci-badge]][dependencyci]
[![Make a pull request][prs-badge]][prs]
[![License](http://img.shields.io/badge/Licence-MIT-brightgreen.svg)](LICENSE.md)

[![Watch on GitHub][github-watch-badge]][github-watch]
[![Star on GitHub][github-star-badge]][github-star]
[![Tweet][twitter-badge]][twitter]

# Introduction

Bootstrap and package your project with Angular 8 and Electron (Typescript + SASS + Hot Reload) for creating Desktop applications.

Currently runs with:

- Angular v8.1.3
- Electron v6.0.2
- Electron Builder v21.2.0

With this sample, you can :

- Run your app in a local development environment with Electron & Hot reload
- Run your app in a production environment
- Package your app into an executable file for Linux, Windows & Mac

/!\ Angular 8.0 CLI needs Node 10.9 or later to work.

## Getting Started

Clone this repository locally :

``` bash
git clone https://github.com/maximegris/angular-electron.git
```

Install dependencies with npm :

``` bash
npm install
```

There is an issue with `yarn` and `node_modules` that are only used in electron on the backend when the application is built by the packager. Please use `npm` as dependencies manager.


If you want to generate Angular components with Angular-cli , you **MUST** install `@angular/cli` in npm global context.
Please follow [Angular-cli documentation](https://github.com/angular/angular-cli) if you had installed a previous version of `angular-cli`.

``` bash
npm install -g @angular/cli
```

## To build for development

- **in a terminal window** -> npm start

Voila! You can use your Angular + Electron app in a local development environment with hot reload !

The application code is managed by `main.ts`. In this sample, the app runs with a simple Angular App (http://localhost:4200) and an Electron window.
The Angular component contains an example of Electron and NodeJS native lib import.
You can disable "Developer Tools" by commenting `win.webContents.openDevTools();` in `main.ts`.

## Included Commands

|Command|Description|
|--|--|
|`npm run ng:serve:web`| Execute the app in the browser |
|`npm run build`| Build the app. Your built files are in the /dist folder. |
|`npm run build:prod`| Build the app with Angular aot. Your built files are in the /dist folder. |
|`npm run electron:local`| Builds your application and start electron
|`npm run electron:linux`| Builds your application and creates an app consumable on linux system |
|`npm run electron:windows`| On a Windows OS, builds your application and creates an app consumable in windows 32/64 bit systems |
|`npm run electron:mac`|  On a MAC OS, builds your application and generates a `.app` file of your application that can be run on Mac |

**Your application is optimised. Only /dist folder and node dependencies are included in the executable.**

## You want to use a specific lib (like rxjs) in electron main thread ?

You can do this! Just by importing your library in npm dependencies (not devDependencies) with `npm install --save`. It will be loaded by electron during build phase and added to the final package. Then use your library by importing it in `main.ts` file. Easy no ?

## Browser mode

Maybe you want to execute the application in the browser with hot reload ? You can do it with `npm run ng:serve:web`.
**Note that you can't use Electron or NodeJS native libraries in this case.** Please check `providers/electron.service.ts` to watch how conditional import of electron/Native libraries is done.

## Branch & Packages version

- Angular 4 & Electron 1 : Branch [angular4](https://github.com/maximegris/angular-electron/tree/angular4)
- Angular 5 & Electron 1 : Branch [angular5](https://github.com/maximegris/angular-electron/tree/angular5)
- Angular 6 & Electron 3 : Branch [angular6](https://github.com/maximegris/angular-electron/tree/angular6)
- Angular 7 & Electron 3 : Branch [angular7](https://github.com/maximegris/angular-electron/tree/angular7)
- Angular 8 & Electron 6 : (master)

[build-badge]: https://travis-ci.org/maximegris/angular-electron.svg?branch=master
[build]: https://travis-ci.org/maximegris/angular-electron
[dependencyci-badge]: https://dependencyci.com/github/maximegris/angular-electron/badge
[dependencyci]: https://dependencyci.com/github/maximegris/angular-electron
[license-badge]: https://img.shields.io/badge/license-Apache2-blue.svg?style=flat
[license]: https://github.com/maximegris/angular-electron/blob/master/LICENSE.md
[prs-badge]: https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square
[prs]: http://makeapullrequest.com
[github-watch-badge]: https://img.shields.io/github/watchers/maximegris/angular-electron.svg?style=social
[github-watch]: https://github.com/maximegris/angular-electron/watchers
[github-star-badge]: https://img.shields.io/github/stars/maximegris/angular-electron.svg?style=social
[github-star]: https://github.com/maximegris/angular-electron/stargazers
[twitter]: https://twitter.com/intent/tweet?text=Check%20out%20angular-electron!%20https://github.com/maximegris/angular-electron%20%F0%9F%91%8D
[twitter-badge]: https://img.shields.io/twitter/url/https/github.com/maximegris/angular-electron.svg?style=social
# appAtendimentoExe
# appPlantaoFiscal


CHANGELOG.md
## 6.1.0 (2019-08-15)

* [DELETE]: vscode settings removed ([becc9b4](https://github.com/maximegris/angular-electron/commit/becc9b4))
* [UPDATE]: Typescript version fixes ([a284c23](https://github.com/maximegris/angular-electron/commit/a284c23))
* Angular src restructured as modular as per angular official guidelines, .travis.yml support added fo ([5983703](https://github.com/maximegris/angular-electron/commit/5983703))
* Corejs path updates ([88056d6](https://github.com/maximegris/angular-electron/commit/88056d6))
* Electron 6.0.0 ([1aef350](https://github.com/maximegris/angular-electron/commit/1aef350))
* fix/ Version Electron in README ([acf0d4f](https://github.com/maximegris/angular-electron/commit/acf0d4f))
* misc/ upgrade Angular 8.2 / Spectron 7 ([6e2211f](https://github.com/maximegris/angular-electron/commit/6e2211f))
* Update dependencies ([bd51f28](https://github.com/maximegris/angular-electron/commit/bd51f28))



## <small>6.0.1 (2019-05-31)</small>

* [Bumped Version] 6.0.1 ([4f9cef3](https://github.com/maximegris/angular-electron/commit/4f9cef3))
* ref/ strict version build-angular & zonejs ([5a24b56](https://github.com/maximegris/angular-electron/commit/5a24b56))
* ref/ strict version codelyzer ([6ede0c8](https://github.com/maximegris/angular-electron/commit/6ede0c8))



## 6.0.0 (2019-05-31)

* [Bumped Version] 6.0.0 ([fb719ab](https://github.com/maximegris/angular-electron/commit/fb719ab))
* Add CI for macOS. ([3ba02e3](https://github.com/maximegris/angular-electron/commit/3ba02e3))
* bump angular version ([7a564a0](https://github.com/maximegris/angular-electron/commit/7a564a0))
* bump tslint & codelyzer versions, update tslint rules & alphabetize ([8425cdc](https://github.com/maximegris/angular-electron/commit/8425cdc))
* Fix Travis CI link ([10aaa4c](https://github.com/maximegris/angular-electron/commit/10aaa4c))
* Not gitignore src/karma.conf.js ([7599320](https://github.com/maximegris/angular-electron/commit/7599320))
* ref/ upgrade Angular 8 and Electron 5 ([92334cf](https://github.com/maximegris/angular-electron/commit/92334cf))
* Remove Node.js v12. ([ccbf6cd](https://github.com/maximegris/angular-electron/commit/ccbf6cd))
* Run CI for Node.js version 10 ([2538965](https://github.com/maximegris/angular-electron/commit/2538965))
* update versions and prepare for electron 5 ([07a5786](https://github.com/maximegris/angular-electron/commit/07a5786))
* Use service xvfb. ([4db31e3](https://github.com/maximegris/angular-electron/commit/4db31e3))
* version bump ([bb1d6bb](https://github.com/maximegris/angular-electron/commit/bb1d6bb))
* feat(CI): update Ubuntu and Node versions in Travis ([e0ff557](https://github.com/maximegris/angular-electron/commit/e0ff557))
* fix: type conflicts ([a2971bf](https://github.com/maximegris/angular-electron/commit/a2971bf))
* fix(e2e): add mocha types ([20e1e89](https://github.com/maximegris/angular-electron/commit/20e1e89))
* fix(e2e): without devTools ([2581983](https://github.com/maximegris/angular-electron/commit/2581983)), closes [/github.com/electron/spectron/issues/174#issuecomment-319242097](https://github.com//github.com/electron/spectron/issues/174/issues/issuecomment-319242097)
* chore: Spectron for e2e tests ([901438a](https://github.com/maximegris/angular-electron/commit/901438a))



## 5.1.0 (2018-11-30)

* [Bumped Version] 5.1.0 ([b790e12](https://github.com/maximegris/angular-electron/commit/b790e12))
* fix/ typo Angular 7 ([fde371f](https://github.com/maximegris/angular-electron/commit/fde371f))
* fix/ typo README ([723233c](https://github.com/maximegris/angular-electron/commit/723233c))
* fix/ typo script npm electron:windows ([45bab44](https://github.com/maximegris/angular-electron/commit/45bab44))
* ref/ remve npx - fix vulnerabilities ([41aeb57](https://github.com/maximegris/angular-electron/commit/41aeb57))
* update README ([f146d5d](https://github.com/maximegris/angular-electron/commit/f146d5d))



## 5.0.0 (2018-11-11)

* Fix typos in README file ([0440ee9](https://github.com/maximegris/angular-electron/commit/0440ee9))
* ref/ Generate changelog ([a89b3ce](https://github.com/maximegris/angular-electron/commit/a89b3ce))
* ref/ Upgrade to Angular 7 ([315a79b](https://github.com/maximegris/angular-electron/commit/315a79b))
* Update electron-builder.json files rule ([82c7bcf](https://github.com/maximegris/angular-electron/commit/82c7bcf))
* Update Version Electron 2 to 3 #hacktoberfest ([f083328](https://github.com/maximegris/angular-electron/commit/f083328))



## <small>4.2.2 (2018-08-22)</small>

* fix/ build serve & electron with single tsc command ([9106c8f](https://github.com/maximegris/angular-electron/commit/9106c8f))
* fix/ typo README ([a9448aa](https://github.com/maximegris/angular-electron/commit/a9448aa))



## <small>4.2.1 (2018-08-22)</small>

* fix/ jslib in main process error ([ef33f5e](https://github.com/maximegris/angular-electron/commit/ef33f5e))



## 4.2.0 (2018-08-19)

* [Bumped Version] V4.2.0 ([0da3856](https://github.com/maximegris/angular-electron/commit/0da3856))
* fix/ electron builder output directories #200 ([f4535e5](https://github.com/maximegris/angular-electron/commit/f4535e5)), closes [#200](https://github.com/maximegris/angular-electron/issues/200)
* Make sure tsconfig be used. ([961c8b1](https://github.com/maximegris/angular-electron/commit/961c8b1))
* ref/ remove some directories of tsconfig.app.json ([1adad4a](https://github.com/maximegris/angular-electron/commit/1adad4a))
* Upgrade Angular (6.1.2) deps ([d8818c1](https://github.com/maximegris/angular-electron/commit/d8818c1))



## 4.1.0 (2018-06-27)

* Allow Angular Using Electron Modules ([ec705ee](https://github.com/maximegris/angular-electron/commit/ec705ee))
* fix/ version angular (revert 6.0.6 -> 6.0.5) ([63a41b8](https://github.com/maximegris/angular-electron/commit/63a41b8))

src/app/models/grupo/grupo.ts
import { IGrupo } from '../../interfaces/grupo/igrupo';
export class Grupo {
  getGrupos(): IGrupo[] {
    return [
      { tipo: 'ACESSÓRIOS AUTOMOTIVOS', grupo: '01' },
      { tipo: 'ACESSÓRIOS DE INFORMÁTICA', grupo: '02' },
      { tipo: 'ARTESANATO/SUVINIR', grupo: '03' },
      { tipo: 'BARRACA PADRÃO', grupo: '04' },
      { tipo: 'BEBIDAS ALCOÓLICAS', grupo: '05' },
      { tipo: 'BEBIDAS EM GARRFAS DE VIDRO', grupo: '06' },
      { tipo: 'BEBIDAS NÃO ALCOÓLICAS', grupo: '07' },
      { tipo: 'BICICLETA', grupo: '08' },
      { tipo: 'BOLSAS E MOCHILAS', grupo: '09' },
      { tipo: 'BOTIJÃO DE GÁS', grupo: '10' },
      { tipo: 'BRINQUEDOS', grupo: '11' },
      { tipo: 'BURRO SEM RABO', grupo: '12' },
      { tipo: 'CADEIRA', grupo: '13' },
      { tipo: 'CALÇADO', grupo: '14' },
      { tipo: 'CARRINHO DE CARGA', grupo: '15' },
      { tipo: 'CHURRASQUEIRA', grupo: '16' },
      { tipo: 'ELETRODOMÉSTICO', grupo: '17' },
      { tipo: 'ELETRÔNICOS', grupo: '18' },
      { tipo: 'ENGENHO PUBLICITÁRIO', grupo: '19' },
      { tipo: 'GRANDES ESTRUTURAS', grupo: '20' },
      { tipo: 'GUARDA CHUVA', grupo: '21' },
      { tipo: 'ISOPOR', grupo: '22' },
      { tipo: 'LIVRO', grupo: '23' },
      { tipo: 'MATERIAL DE PRAIA', grupo: '24' },
      { tipo: 'MESA', grupo: '25' },
      { tipo: 'MIDIA', grupo: '26' },
      { tipo: 'MOENDA', grupo: '27' },
      { tipo: 'OCULOS', grupo: '28' },
      { tipo: 'PAPELARIA E BAZAR', grupo: '29' },
      { tipo: 'PERECÍVEIS', grupo: '30' },
      { tipo: 'QUENTINHA', grupo: '31' },
      { tipo: 'QUINQUILHARIA', grupo: '32' },
      { tipo: 'TABULEIRO', grupo: '33' },
      { tipo: 'TRICÍCLO', grupo: '34' },
      { tipo: 'UTENSÍLIOS DE CARROÇA', grupo: '35' },
      { tipo: 'VEÍCULO AUTOMOTOR', grupo: '36' },
      { tipo: 'VESTUÁRIO', grupo: '37' }
    ];
  }

  getTipo(grupo: string): string {
    const index = this.getGrupos().findIndex(x => x.grupo === grupo);
    return this.getGrupos()[index].tipo;
  }
}
